compton: 0.1_beta2.5 -> 6.2

JohnAZoidberg opened the pull request:
###### Motivation for this change
Many new releases:
- https://github.com/yshui/compton/releases/tag/v5.1
- https://github.com/yshui/compton/releases/tree/v6
- https://github.com/yshui/compton/releases/tree/v6.2

Maintainers: @ertes @enzime @twey

###### Things done

I'm using it with the NixOS module and its default settings. Seems fine so far, running all day today.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

JohnAZoidberg committed:
compton-git: 5.1-rc2 -> 6.2

vsync is now a boolean option, see:
https://github.com/yshui/compton/pull/130

menu-opacity is deprecated and there's a warning that says:
Please use the wintype option `opacity` of `popup_menu` and
`dropdown_menu` instead.

rycee commented on nixos/modules/services/x11/compton.nix:
```diff
@@ -189,15 +194,11 @@ in {
     };
 
     vSync = mkOption {
-      type = types.enum [
```
To maintain backwards compatibility I'd suggest having

    with types; either bool (enum [ "none" … "opengl-mswc" ])

instead and then adding something like
```
apply = x:
  let
    res = x != "none";
    msg =
      "The type of services.compton.vSync has changed to bool:"
      + " interpreting ${x} as ${res}";
  in
    if isBool x then x
    else warn msg res;
```

JohnAZoidberg replied:
Alright, did that as a separate commit so that we can revert it easily in the future when it's reasonable to assume everyone has upgraded.

rycee replied:
Nice! But I was thinking of using the actual [`apply` function](https://github.com/NixOS/nixpkgs/blob/6d3b7390bc683eea884f9e1d5846632e4fe10428/lib/options.nix#L43-L44) of the `mkOption` argument.

See, for example, in `system-environment.nix`: https://github.com/NixOS/nixpkgs/blob/6d3b7390bc683eea884f9e1d5846632e4fe10428/nixos/modules/config/system-environment.nix#L17-L28

JohnAZoidberg replied:
Ohh, thanks! I didn't know that

JohnAZoidberg committed:
nixos/compton: Keep vSync option backwards compatible

The new upstream option tries to make the best choice for the user.
Therefore the behaviour should stay the same with this backwards
compatibility patch.

JohnAZoidberg committed:
compton-git: Remove DRM option

It's deprecated and shouldn't be used.
https://github.com/yshui/compton/pull/130/files#r285505456

JohnAZoidberg committed:
compton-git: Remove new_backends option

Was removed in "Let old/new backends co-exist"
b0c5db9f5aa500dc3568cc6fe68493df98794d4d

c0bw3b commented:
The module is using the old `compton` and not `compton-git` by default?
I guess it will be hard to maintain a common module if they start using distinct options...

IMHO we should keep only the maintained fork from `yshui` repo and call it `compton`
It's already the source for compton pkg in Arch / Mageia / Void for example.

JohnAZoidberg committed:
compton: 0.1_beta2.5 -> 6.2

Drop the legacy, unmaintained version and use the fork for real.

JohnAZoidberg commented:
Yes, that seems reasonable.

c0bw3b commented:
Add an alias `compton-git = compton` to avoid breaking existing users' configs

Ma27 reviewed (approved):
The change (and the measures to retain backwards compatibility) seem fine :+1: 

Tested against my local compton config (which is basically the default config with two opacity rules).

c0bw3b commented:
Ah nice to have a feedback from an actual compton user, thanks @Ma27

Infinisil commented:
On a related note, I'm excited for https://github.com/yshui/compton/issues/32

eadwu commented:
Effectively breaks some configs since compton doesn't seem to take well of duplicate settings.

JohnAZoidberg commented:
Oh you mean if you have `wintypes` in your `extraConfig`?
What do you have there?

Infinisil commented:
The idea in NixOS/rfcs#42 would solve this rather well

eadwu commented:
Yes, for example in `compton.sample.conf` from the repository. `extraWintypesConfig` would fix it.

```ini
wintypes:
{
  tooltip = { fade = true; shadow = true; opacity = 0.75; focus = true; full-shadow = false; };
  dock = { shadow = false; }
  dnd = { shadow = false; }
  popup_menu = { opacity = 0.8; }
  dropdown_menu = { opacity = 0.8; }
};
```
